### PR TITLE
Enable Software TPM or Discrete TPM to Windows

### PR DESCRIPTION
**Pull request by christopherco**
This patch adds advertises a Software TPM or a Discrete TPM to Windows.
Selection is done at compile time, behind a build config flag in the board's
.dsc file.

We add two new configuration flags to the Pi2 and Pi3 Board .dsc file.
* `CONFIG_TPM_ENABLE` controls whether to advertise any TPM support to Windows.
* `CONFIG_USE_DTPM` enables the use of the discrete TPM, otherwise use
the software TPM. Note: this option requires `CONFIG_TPM_ENABLE = TRUE`.

Advertising a TPM requires two components - the TPM2 table
and device nodes. The TPM2 table advertises a TPM2 module
is present. We add the TPM device nodes by adding an SSDT
to extend the DSDT table.

Note: Software TPM is emulation and, as such, is not
fit for production security use.

### CODE REVIEW

File: Pi2BoardPkg/AcpiTables/Common/Tpm2DeviceSpi0.asl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        Device(TPM1)
+        {
+            Name(_HID, "MSFT0101")
+            Name(_UID, 1)
+            Name(_DDN, "SPI TPM 2.0 Device")            // _DDN: DOS Device Name
+            Name(_STR, Unicode ("SPI TPM 2.0 Device"))  // _STR: Description String
+            Name (RBUF, ResourceTemplate ()
+            {
+                // RaspberryPi2 4MHz, SPI0/CS0
```
Comment (hbeberman) [inline Pi2BoardPkg/AcpiTables/Common/Tpm2DeviceSpi0.asl]: nit: This file is for the Pi3 as well right?

File: Pi2BoardPkg/AcpiTables/Common/SoftwareTpm2Table.aslc [hunk 2]
```diff
@@ -0,0 +1,46 @@
+//  Copyright (c), Microsoft Corporation. All rights reserved.
```
Comment (hbeberman) [inline Pi2BoardPkg/AcpiTables/Common/SoftwareTpm2Table.aslc:1]: nit: Slight inconsistencies in copyright headers. Some have years, some don't.

**Review by hbeberman — APPROVED** [review]

File: Pi2BoardPkg/AcpiTables/Common/Tpm2DeviceSpi0.asl (hunk 1 above)
Comment (christopherco) [reply Pi2BoardPkg/AcpiTables/Common/Tpm2DeviceSpi0.asl]: Should be applicable on all RPIs so long as the header is the same layout.

File: Pi2BoardPkg/AcpiTables/Common/SoftwareTpm2Table.aslc (hunk 2 above)
Comment (christopherco) [reply Pi2BoardPkg/AcpiTables/Common/SoftwareTpm2Table.aslc:1]: Removed years for consistency with most other headers.

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**